pkg/kds/context/context.go

```diff
@@ -69,7 +70,10 @@ func DefaultContext(
 			),
 			MapZoneTokenSigningKeyGlobalToPublicKey),
 		reconcile.If(
-			reconcile.IsKubernetes(cfg.Store.Type),
+			reconcile.And(
+				reconcile.IsKubernetes(cfg.Store.Type),
+				reconcile.Not(reconcile.TypeIs(mesh.ZoneIngressType)),
```

Comment:
Could you help me understand why we need this change?

Reply:
If I noticed that correctly, when we have 2 zones: one old and one new, we can observe that every reconciliation tick zone ingress changes the name:

```
 2024-01-24T16:08:23.305Z    INFO    kds-delta-zone    creating a new resource from upstream    {"type": "ZoneIngress", "name": "zone-251.kuma-ingress-85df4875bd-rnkml.kuma-system", "mesh": ""}                   │
│ 2024-01-24T16:08:24.292Z    INFO    kds-delta-zone    deleting a resource since it's no longer available in the upstream    {"type": "ZoneIngress", "name": "zone-251.kuma-ingress-85df4875bd-rnkml.kuma-system",  │
│ 2024-01-24T16:08:24.299Z    INFO    kds-delta-zone    creating a new resource from upstream    {"type": "ZoneIngress", "name": "zone-251.kuma-ingress-85df4875bd-rnkml.kuma-system.kuma-system", "mesh": ""}       │
│ 2024-01-24T16:08:26.294Z    INFO    kds-delta-zone    deleting a resource since it's no longer available in the upstream    {"type": "ZoneIngress", "name": "zone-251.kuma-ingress-85df4875bd-rnkml.kuma-system.ku │
│ 2024-01-24T16:08:26.310Z    INFO    kds-delta-zone    creating a new resource from upstream    {"type": "ZoneIngress", "name": "zone-251.kuma-ingress-85df4875bd-rnkml.kuma-system", "mesh": ""}```

That causes issue with constant changes being propagated

Reply:
I understand what you see, but I don't understand the cause. I'm not sure that's the right approach. Let me test a thing or two...

Reply:
We did a test and the cause was that we were not cloning an object after trimming a suffix and instead we were changing an object in the resource cache. Each iteration had a different name and that caused the issue